File: app/src/main/java/com/asteph11/finalproject/controllers/fragments/SetupFragment.java
package com.asteph11.finalproject.controllers.fragments;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.navigation.Navigation;

import com.asteph11.finalproject.R;
import com.asteph11.finalproject.models.data.Grid;

public class SetupFragment extends AGridFragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);

        return inflater.inflate(R.layout.fragment_setup, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        Button readyButton = view.findViewById(R.id.readyButton);

        LayoutInflater inflater = LayoutInflater.from(getContext());

        masterGrid = inflater.inflate(
                R.layout.layout_grid,
                (ViewGroup) view,
                false);
        createGrid(view, masterGrid);
        setCoordinateListeners();

        readyButton.setOnClickListener(v -> {
            boolean gameReady = false;
            try {
                gameReady = matchViewModel.playersReady();
            } catch (Exception e) {
                Log.e("Swap", e.getMessage());
            }
            matchViewModel.getTurnHandler().swapTurns();
            if(!gameReady) {
                Navigation.findNavController(v).
                        navigate(R.id.action_setupFragment_to_awaitSwapFragment);
            } else {
                Navigation.findNavController(v).
                        navigate(R.id.action_setupFragment_to_gameFragment);
            }

        });
    }

    protected void setCoordinateListeners() {
        for(int y = 0; y < Grid.DIMENSIONS; y++) {
            for(int x = 0; x < Grid.DIMENSIONS; x++) {
                int finalX = x;
                int finalY = y;
                gridCoords[y][x].setOnClickListener(view -> {
                    matchViewModel.getTurnHandler().getCurrentPlayer().setShip(finalX, finalY);
                    ImageView status = gridCoords[finalY][finalX].findViewById(R.id.status);
                    status.setImageDrawable(getResources().getDrawable(R.drawable.ship, getActivity().getTheme()));
                    status.setVisibility(View.VISIBLE);
                    try {
                        System.out.println(matchViewModel.getTurnHandler().getCurrentPlayer().getGrid());
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                });
            }
        }
    }
}

File: app/src/main/java/com/asteph11/finalproject/controllers/fragments/SetupFragment.kt
package com.asteph11.finalproject.controllers.fragments

import androidx.navigation.Navigation.findNavController
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.util.Log
import android.view.View
import com.asteph11.finalproject.R
import androidx.cardview.widget.CardView
import android.widget.EditText
import android.widget.ImageView
import com.asteph11.finalproject.models.data.Grid
import java.lang.Exception

class SetupFragment : AGridFragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        return inflater.inflate(R.layout.fragment_setup, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val readyButton = view.findViewById<CardView>(R.id.readyButton)
        val nameInput = view.findViewById<EditText>(R.id.nameInput)
        val inflater = LayoutInflater.from(context)
        masterGrid = inflater.inflate(
            R.layout.layout_grid,
            view as ViewGroup,
            false
        )
        createGrid(view, masterGrid!!)
        setCoordinateListeners()
        readyButton.setOnClickListener { v: View? ->
            if (nameInput.text.toString().equals("", ignoreCase = true) ||
                !matchViewModel?.turnHandler!!.currentPlayer!!.isGridReady
            ) {
                return@setOnClickListener
            }
            var gameReady = false
            try {
                gameReady = matchViewModel?.playersReady()!!
            } catch (e: Exception) {
                Log.e("Swap", e.message!!)
            }
            matchViewModel?.turnHandler!!.swapTurns()
            if (!gameReady) {
                matchViewModel?.turnHandler!!.otherPlayer!!.name = nameInput.text.toString()
                nameInput.setText("")
                findNavController(v!!).navigate(R.id.action_setupFragment_to_awaitSwapFragment)
            } else {
                matchViewModel?.turnHandler!!.otherPlayer!!.name = nameInput.text.toString()
                findNavController(v!!).navigate(R.id.action_setupFragment_to_gameFragment)
            }
        }
    }

    override fun setCoordinateListeners() {
        for (y in 0 until Grid.DIMENSIONS) {
            for (x in 0 until Grid.DIMENSIONS) {
                gridCoords[y][x]?.setOnClickListener { view: View? ->
                    val status = gridCoords[y][x]?.findViewById<ImageView>(R.id.status)
                    matchViewModel?.turnHandler!!.currentPlayer!!.setShip(x, y)
                    if (matchViewModel?.turnHandler!!.currentPlayer!!.grid!!.getStatusAt(
                            x,
                            y
                        ) === Grid.Status.SHIP
                    ) {
                        status?.setImageDrawable(
                            resources.getDrawable(
                                R.drawable.ship,
                                requireActivity().theme
                            )
                        )
                        status?.visibility = View.VISIBLE
                    } else {
                        status?.setImageDrawable(null)
                        status?.visibility = View.INVISIBLE
                    }
                    try {
                        println(matchViewModel?.turnHandler!!.currentPlayer!!.grid)
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }
            }
        }
    }
}